platforms/enterprise/enterprise-operations/src/main/java/org/gradle/operations/configuration/ConfigurationCacheCheckFingerprintBuildOperationType.java

```diff
@@ -0,0 +1,179 @@
+         * <p>
+         * Not all invalidation reasons may be returned. However, if the entry is invalidated, at least one reason is listed.
+         * <p>
+         * The first invalidation of the first build in the list is what Gradle shows to the user.
+         *
+         * @return the list of invalidation reasons
+         * @since 8.10
+         */
+        List<BuildInvalidationReasons> getBuildInvalidationReasons();
+
+        /**
+         * Returns a list of project-level invalidation reasons. Only contains entries for projects that were invalidated.
+         * This list is empty if the whole build is invalidated, i.e. when {@link #getBuildInvalidationReasons()} returns a non-empty list.
+         * <p>
+         * All invalidated project have entries in this list, however, not all invalidation reasons may be listed.
+         * <p>
+         * The first invalidation of the first project in the list is what Gradle shows to the user.
+         * The rest of the invalidated projects follows, ordered by the {@link ProjectInvalidationReasons#getIdentityPath()}.
+         *
+         * @return a list of invalidation projects with their invalidation reasons
+         * @since 8.10
+         */
+        List<ProjectInvalidationReasons> getProjectInvalidationReasons();
+    }
+
+    /**
+     * Contains invalidation for a single build.
+     *
+     * @since 8.10
+     */
+    public interface BuildInvalidationReasons {
+        /**
+         * Returns the path of the invalidated build in the build tree.
+         *
+         * @return the build path
+         * @since 8.10
+         */
+        String getBuildPath();
+
+        /**
+         * Returns the list of the invalidation reasons for the given build. At least one invalidation is always present.
+         *
+         * @return the list of invalidation reasons
+         * @since 8.10
+         */
+        List<FingerprintInvalidationReason> getInvalidationReasons();
+    }
+
+    /**
+     * Contains data of a single invalidated project.
+     *
+     * @since 8.10
+     */
+    public interface ProjectInvalidationReasons {
+        /**
+         * Returns the identity path of the invalidated project.
+         *
+         * @return the identity path
+         * @since 8.10
+         */
+        String getIdentityPath();
+
+        /**
+         * Returns the list of invalidation reasons. At least one invalidation is always present.
+         *
+         * @return the list of invalidation reasons
+         * @since 8.10
+         */
+        List<FingerprintInvalidationReason> getInvalidationReasons();
+    }
+
+    /**
+     * Represents the invalidation reason.
+     *
+     * @since 8.10
+     */
+    public interface FingerprintInvalidationReason {
+        /**
+         * Returns a simple text representation of the invalidation reason.
+         *
+         * @return the invalidation reason as text
+         * @since 8.10
+         */
+        String getMessage();
```

Comment:
🌈  It would be nice if we could return some sort of `StructuredMessage` annotated with information that would allow consumers to understand references to projects, system properties, etc, rather than having to resort to parsing based on conventions.